doc/netplan.md

```diff
@@ -678,6 +678,67 @@ wpasupplicant installed if you let the ``networkd`` renderer handle wifi.
           ``active-backup``, ``balance-alb``, and ``balance-tlb`` modes.
 
 
+## Properties for device type ``tunnels:``
+
+``parent`` (scalar)
+
+:    The parent device for the tunnel (optional).
+
+``mode`` (scalar)
+
+:   Defines the tunnel mode. Valid options are ``sit``, ``gre``, ``ip6gre``,
+    ``ipip``, ``ipip6``, ``ip6ip6``, ``vti``, and ``vti6``. Additionally,
+    the ``networkd`` backend also supports ``gretap`` and ``ip6gretap`` modes.
+    In addition, the ``NetworkManager`` supports ``isatap`` tunnels.
+
+``local`` (scalar)
+
+:   Defines the address of the local endpoint of the tunnel.
+
+``remote`` (scalar)
+
+:   Defines the address of the remote endpoint of the tunnel.
+
+``key``  (scalar) or (mapping)
+:   See ``keys`` below.
```

Comment:
Either way around works. We do already support another typo due to its introduction in syntax previously.

I can rearrange doc, this makes no difference in the actual implementation; given that the same field can be both a scalar or a mapping, for ease of use and comprehension by the user, "key" and "keys" have been made equivalent. I want to avoid people being tripped because they used "keys", when they should have used "key" despite the use of a mapping, and "keys" when they only specify a single string.